libs/partners/ai21/langchain_ai21/semantic_text_splitter.py

```diff
@@ -0,0 +1,132 @@
+import copy
+import logging
+import re
+from typing import (
+    Any,
+    Iterable,
+    List,
+    Optional,
+)
+
+from ai21.models import DocumentType
+from langchain_core.documents import Document
+from langchain_text_splitters import TextSplitter
+
+from langchain_ai21.ai21_base import AI21Base
+
+logger = logging.getLogger(__name__)
+
+
+class AI21SemanticTextSplitter(TextSplitter):
+    """Splitting text into coherent and readable units,
+    based on distinct topics and lines
+    """
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        arbitrary_types_allowed = True
+
+    def __init__(
+        self,
+        chunk_size: int = 0,
+        chunk_overlap: int = 0,
+        client: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Create a new TextSplitter."""
+        super().__init__(
+            chunk_size=chunk_size,
+            chunk_overlap=chunk_overlap,
+            **kwargs,
+        )
+
+        self._ai21_base = AI21Base(client=client)
```

Comment:
I think instead of `self._ai21_base` it'd be more accurate to create `self._segmentation = AI21Base().client.segmentation` instead, as this is the object we need and not the base. wdyt?

Reply:
In our other component we didn't only create the relevant object, so I tried to keep the convention as much as possible (besides the limitation that I had that I couldn't inherit from AI21Base).
Either way, it's not exposed as part of the class, so it won't affect the users, and it's the simplest implementation.